File: BookStoreNetReact.Domain/Entities/Author.cs
namespace BookStoreNetReact.Domain.Entities
{
    public class Author : BaseEntity
    {
        public required string FullName { get; set; }
        public required string Biography { get; set; }
        public required string Country { get; set; }
        public string? PublicId { get; set; }
        public string? ImageUrl { get; set; }
        public List<Book> Books { get; set; } = new List<Book>();
    }
}

File: client/src/app/models/author.ts
import { FilterParams } from "./pagination";

export interface Author {
    id: number;
    fullName: string;
    country: string;
    publicId?: string;
    imageUrl?: string;
}

export interface AuthorDetail extends Author {
    biography: string;
}

export interface AuthorParams extends FilterParams {
    countries: string[];
}
